src/api.ts

```diff
@@ -2110,41 +2117,74 @@ export class API {
   }
 
   /**
-   * Retrieves a prompt by its id.
-   *
-   * @param id ID of the prompt to retrieve.
-   * @returns The prompt with given ID.
+   * Retrieves a prompt by its id. If the request fails, it will try to get the prompt from the cache.
    */
   public async getPromptById(id: string) {
     const query = `
-    query GetPrompt($id: String!) {
-      promptVersion(id: $id) {
-        createdAt
-        id
-        label
-        settings
-        status
-        tags
-        templateMessages
-        tools
-        type
-        updatedAt
-        url
-        variables
-        variablesDefaultValues
-        version
-        lineage {
-          name
+      query GetPrompt($id: String!) {
+        promptVersion(id: $id) {
+          createdAt
+          id
+          label
+          settings
+          status
+          tags
+          templateMessages
+          tools
+          type
+          updatedAt
+          url
+          variables
+          variablesDefaultValues
+          version
+          lineage {
+            name
+          }
         }
       }
-    }
     `;
 
     return await this.getPromptWithQuery(query, { id });
   }
 
   /**
-   * Retrieves a prompt by its name and optionally by its version.
+   * Private helper method to execute prompt queries with error handling and caching
+   */
+  private async getPromptWithQuery(
+    query: string,
+    variables: Record<string, any>
```

Comment:
Let's improve the typing here, we know the key we expect:
```suggestion
    variables: { id?: string, name?: string, version?: string }
```